neurom/view/common.py

```diff
@@ -596,16 +591,94 @@ def plot_legend(ax, **kwargs):
         ax.legend(**legend_arg)
 
 
-def plot_sphere(_, ax, center, radius, color='black', alpha=1.):
-    """Plots a 3d sphere, given the center and the radius."""
+_LINSPACE_COUNT = 300
 
-    u = np.linspace(0, 2 * np.pi, 300)
-    v = np.linspace(0, np.pi, 300)
 
-    x = center[0] + radius * np.outer(np.cos(u), np.sin(v))
-    y = center[1] + radius * np.outer(np.sin(u), np.sin(v))
+def _get_normals(v):
+    '''get two vectors that form a basis w/ v
+
+    Note: returned vectors are unit
+    '''
+    not_v = np.array([1, 0, 0])
+    if np.all(np.abs(v) == not_v):
+        not_v = np.array([0, 1, 0])
+    n1 = np.cross(v, not_v)
+    n1 /= norm(n1)
+    n2 = np.cross(v, n1)
+    return n1, n2
+
+
+def generate_cylindrical_points(start, end, start_radius, end_radius,
+                                linspace_count=_LINSPACE_COUNT):
+    '''Generate a 3d mesh of a cylinder with start and end points, and varying radius
+
+    Based on: http://stackoverflow.com/a/32383775
+    '''
+    v = end - start
+    length = norm(v)
+    v = v / length
+    n1, n2 = _get_normals(v)
+
+    # pylint: disable=unbalanced-tuple-unpacking
+    l, theta = np.meshgrid(np.linspace(0, length, linspace_count),
+                           np.linspace(0, 2 * np.pi, linspace_count))
+
+    radii = np.linspace(start_radius, end_radius, linspace_count)
+    rsin = np.multiply(radii, np.sin(theta))
+    rcos = np.multiply(radii, np.cos(theta))
+
+    return np.array([start[i] +
+                     v[i] * l +
+                     n1[i] * rsin + n2[i] * rcos
+                     for i in range(3)])
+
+
+def project_cylinder_onto_2d(ax, plane,
+                             start, end, start_radius, end_radius,
+                             color='black', alpha=1.):
+    '''take cylinder defined by start/end, and project it onto the plane
+
+    Args:
+        ax: matplotlib axes
+        plane(tuple of int): where x, y, z = 0, 1, 2, so (0, 1) is the xy axis
+        start(np.array): start coordinates
+        end(np.array): end coordinates
+        start_radius(float): start radius
+        end_radius(float): end radius
+        color: matplotlib color
+        alpha(float): alpha value
+
+    Note: There are probably more efficient ways of doing this: here the
```

Comment:
It looks reasonable to me! Is this plotting generalized to a collection of cylinders if the swc-soma is a set of cylinders?